.github/workflows/ci.yml

```diff
@@ -32,7 +32,11 @@ jobs:
           fi
 
       - name: Install dependencies
-        run: pip install -r requirements.txt
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+          cache: 'pip'
```

Comment:
Does this work? I tried caching some stuff and it didn't make a difference at all in the build times. Seemed a bit random. Curious if it works for you. Thanks.